include/pybind11/pybind11.h

```diff
@@ -216,9 +214,8 @@ class cpp_function : public function {
                 const std::type_info *t = types[type_index++];
                 if (!t)
                     pybind11_fail("Internal error while parsing type signature (1)");
-                auto it = registered_types.find(std::type_index(*t));
```

Comment:
Indeed, that one is. I remember seeing something in ``type_caster_generic`` though.